### PR TITLE
#54 - Manifest abstraction added

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #54 
Added Manifest interfaces and JsonManifest implementation with methods to read type and save manifest content.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#79](https://github.com/artipie/docker-adapter/issues/79) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/92bb2487-236a-48c7-b898-bf402fc83f46); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/docker/manifest/JsonManifest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public Content content() {
+        return this.source;
+    }
+
+    /**
+     * Reads content as JSON object.
+     *
+     * @return JSON object.
+     */
+    private CompletionStage<JsonObject> json() {
```
Comment (paulodamaso) [inline src/main/java/com/artipie/docker/manifest/JsonManifest.java:72]: @olegmoz I think that you could extract this private method to a class; seems like we will need this Content - Json transformation more than once in this project

Comment (olegmoz) [reply src/main/java/com/artipie/docker/manifest/JsonManifest.java]: @paulodamaso good idea, done

**Comment by paulodamaso** [comment]
@olegmoz Great, thanks!

**Comment by paulodamaso** [comment]
@g4s8 ping

**Comment by 0crat** [comment]
Job was finished in 45 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@paulodamaso please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@paulodamaso ping

**Comment by paulodamaso** [comment]
@sereshqua I confirm, thank you

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**